### PR TITLE
refactor(Login screen): Move login screen to Formik

### PR DESCRIPTION
**Pull request by jamesbrooks94**
Fixes https://github.com/zoe/covid-tracker-react-native/issues/90

### CODE REVIEW

**Comment by jamesbrooks94** [comment]
Includes fixes for https://github.com/zoe/covid-tracker-react-native/issues/89

File: src/features/login/LoginScreen.tsx [hunk 1]
```diff
@@ -1,193 +1,251 @@
...
-    private errorMessage = ""
-    private username = ""
-    private password = ""
-
-    constructor(props: PropsType) {
-        super(props);
-        this.state = initialState;
-        this.handleLogin = this.handleLogin.bind(this)
-    }
+const RegisterStartLink: React.FC<PropsType> = ({ navigation }) => {
```
Comment (julienlavigne) [inline src/features/login/LoginScreen.tsx:40]: Shall navigation be typed?

File: src/features/login/LoginScreen.tsx [hunk 2]
```diff
@@ -1,193 +1,251 @@
...
+                        name: getWelcomeRepeatScreenName(),
+                        params: { patientId },
+                    },
+                ],
+            });
+        } catch (error) {
+            const errorMessage = i18n.t(
+                error.constructor === UserNotFoundException
+                    ? "login.user-not-found-exception"
+                    : "login.exception"
```
Comment (julienlavigne) [inline src/features/login/LoginScreen.tsx:91]: Nitpick: would extract in a variable

File: src/features/login/LoginScreen.tsx [hunk 3]
```diff
@@ -1,193 +1,251 @@
...
+    return (
+        <Formik
+            onSubmit={handleLogin}
+            initialValues={initialValues}
+            validationSchema={registerSchema}
+            validateOnBlur
+            validateOnChange={false}
+        >
+            {({ handleSubmit, handleChange, values, ...formikProps }) => {
+                console.log(formikProps.touched);
```
Comment (julienlavigne) [inline src/features/login/LoginScreen.tsx:112]: delete?

File: src/features/login/LoginScreen.tsx [hunk 4]
```diff
@@ -1,193 +1,251 @@
...
-                } else {
-                    this.errorMessage = i18n.t("login.exception")
-                }
-                Toast.show({
-                    text: this.errorMessage,
-                    duration: 2500,
-                })
-            });
-    }
+    let inputEl: any | null = null;
```
Comment (julienlavigne) [inline src/features/login/LoginScreen.tsx:101]: Rename to something like passwordInputElement?

File: src/features/login/LoginScreen.tsx [hunk 5]
```diff
@@ -1,193 +1,251 @@
...
+                                            )}
+                                            onBlur={formikProps.handleBlur(
+                                                "password"
+                                            )}
+                                            onSubmitEditing={() => {
+                                                !!inputEl._root &&
+                                                    inputEl._root.focus();
+                                            }}
+                                            blurOnSubmit={false}
+                                        />
```
Comment (julienlavigne) [inline src/features/login/LoginScreen.tsx:166]: I'm confused by that part and would need to pull the code locally to look into it. That might also be my ignorance or react or other, but:
- sounds like the email-address field is duplicated?
- the second onBlur points to the `password` field

File: src/features/patient/AboutYouScreen.tsx [hunk 6]
```diff
@@ -68,71 +63,80 @@ interface AboutYouData {
...
-            gender: formData.sex === 'male' ? 1 : formData.sex === 'female' ? 0 : formData.sex === 'pfnts' ? 2 : 3,
+            year_of_birth: parseInt(formData.yearOfBirth, 10),
+            gender:
+                formData.sex === "male"
+                    ? 1
+                    : formData.sex === "female"
+                    ? 0
+                    : formData.sex === "pfnts"
+                    ? 2
+                    : 3,
```
Comment (julienlavigne) [inline src/features/patient/AboutYouScreen.tsx:139]: 💭 omg... did we write that?!

File: src/features/login/LoginScreen.tsx (hunk 1 above)
Comment (jamesbrooks94) [reply src/features/login/LoginScreen.tsx:40]: Navigation is typed from `: React.FC<PropsType>`

File: src/features/login/LoginScreen.tsx (hunk 5 above)
Comment (jamesbrooks94) [reply src/features/login/LoginScreen.tsx]: Think I managed to mess up a merge - should be resolved now

File: src/features/patient/AboutYouScreen.tsx (hunk 6 above)
Comment (jamesbrooks94) [reply src/features/patient/AboutYouScreen.tsx]: Yep!

**Comment by jamesbrooks94** [comment]
@julienlavigne I have fixed the PR comments and is ready for re-review

**Comment by jamesbrooks94** [comment]
@selvach This is also ready for re-review

**Comment by jamesbrooks94** [comment]
Unsure how I managed to delete this, but it's back :)

**Comment by JoshElgar** [comment]
Sorry for not getting round to reviewing this sooner! In future it would be great to request a review from the 'zoe/covid-mobile' team if you can see it. Would you mind fixing the merge conflicts?

**Comment by jamesbrooks94** [comment]
I’ll fix them tomorrow, I can’t add reviewers it seems, I’ll check on a new PR

**Comment by selvach** [comment]
@jamesbrooks94 - I've opened a new PR based on your changes. https://github.com/zoe/covid-tracker-react-native/pull/227 

I'll close this PR. Thank you very much!

**Pull request closed without merging**